src/libraries/System.Text.Json/tests/Serialization/CodeGenTests/SlowPathTests.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using MyNamespace;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Json;
+using System.Text.Json.Serialization.Metadata;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests.CodeGen
+{
+    public class SlowPathTests
+    {
+        /// <summary>
+        /// When using a Stream, verify that the JsonTypeInfo falls back to standard converters
+        /// instead of calling the generated Serialize and Deserialize methods.
+        /// </summary>
+        /// <returns></returns>
+        [Fact]
+        public static async Task Stream()
+        {
+            const int Count = 100;
+
+            var objs = new List<IndexViewModel>(Count);
+            for (int i = 0; i < Count; i++)
+            {
+                objs.Add(IndexViewModelPocoTests.Create());
+            }
+
+            byte[] utf8 = JsonSerializer.SerializeToUtf8Bytes(objs);
+
+            JsonSerializerOptions options = new JsonSerializerOptions
+            {
+                DefaultBufferSize = 1
+            };
+
+            JsonTypeInfo<List<IndexViewModel>> typeInfo = KnownCollectionTypeInfos<IndexViewModel>.
+                GetList(JsonContext.Default.IndexViewModel, JsonContext.Default);
+
+            using (MemoryStream stream = new MemoryStream(utf8))
+            {
+                List<IndexViewModel> newObjs = await JsonSerializer.DeserializeAsync<List<IndexViewModel>>(
+                    stream,
+                    jsonTypeInfo: typeInfo);
+
+                Assert.Equal(Count, newObjs.Count);
+                for (int i = 0; i < objs.Count; i++)
+                {
+                    IndexViewModelPocoTests.Verify(objs[i], newObjs[i]);
+                }
+            }
+
+            // todo: add SerializeAsync method
```

Comment:
👀

Reply:
As a proof-of-concept, just the DeserializeAsync was added (to ensure callers can still use that, even though the generated (de)serialize methods won't get called. The SerializeAsync will be added next pass.